README.md

```diff
@@ -74,32 +83,60 @@ Notes:
 ## Compiler development
 
 For bash users, adding the following to your `.bashrc` defines some aliases
-that are useful during compiler development, such as shortcuts for builing
+that are useful during compiler development, such as shortcuts for building
 and running `npcomp-opt`.
 
-```
-source $WHERE_YOU_CHECKED_OUT_NPCOMP/tools/bash_helpers.sh
+```shell
+source "${WHERE_YOU_CHECKED_OUT_NPCOMP?}/tools/bash_helpers.sh"
 ```
 
 ## Build Instructions
 
 ### Common prep
 
+Note: If using `pyenv` (or an interpreter manager that depends on it like
+`asdf`), you'll need to use
+[`--enable-shared`](https://github.com/pyenv/pyenv/tree/master/plugins/python-build#building-with---enable-shared)
+during interpreter installation.
+
+#### Install dependencies:
+
 ```shell
-# From checkout directory.
+python3 -m pip install pybind11
+```
+
+Optional: if building with `clang` and `lld`:
```

Comment:
By the way, is this optional, or are `clang` and `lld` required? I believe I forgot `export LDFLAGS=-fuse-ld=$(which "ld.lld-${LLVM_VERSION?}")` at one point, and got some linker errors. It may be possible to work with `gcc` and friends, but perhaps it is simpler to just say that `clang` and `lld` are required?

Reply:
I got the impression that they're optional in the sense that "if you really want to use `gcc` you can try", but I know very little about C/C++ and their compilers. I mostly included this as "optional" because it is noted as such on line 125.

Reply:
That's the impression I got as well. Maybe it should be "recommended" instead of "optional"? Curious what @stellaraccident thinks on this one.

Reply:
Yes, we should make this less of a readout from my shell history :) +1 to "Recommended compiler/linker setup"